charts/signoz/templates/schema-migrator/migrations-init.yaml

```diff
@@ -1,8 +1,10 @@
 {{- if and .Values.schemaMigrator.enabled .Release.IsInstall }}
```

Comment:
@srikanthccv, a couple of suggestions for the stated problem. Full disclosure: I don't have full context of the difference between the sync and the async job.

- We can run the sync job as an init container of the async job. Once the init container exists successfully, the main container (which is the async job) will kick in.

Reply:
The requirement is to make the collector version upgrade depend on the sync job. Once the sync job exits successfully, all the necessary schema changes will have been completed. The async job contains a mutation schema change that can run in the background. We don't want to make the release wait till mutations are complete because they can take a long time to finish.